lib/metrics/context.js

```diff
@@ -43,6 +52,60 @@ module.exports = {
 
     return data
   }
+
+  function validate(request) {
+
+    var metadata = request.payload.metricsContext
+    if (!metadata) {
+      return logInvalidContext(request, 'missing context')
+    }
+    if (!metadata.flowId) {
+      return logInvalidContext(request, 'missing flowId')
+    }
+    if (!metadata.flowBeginTime) {
+      return logInvalidContext(request, 'missing flowBeginTime')
+    }
+
+    if (Date.now() - metadata.flowBeginTime > config.metrics.flow_id_expiry) {
+      return logInvalidContext(request, 'expired flowBeginTime')
+    }
+
+    // The flowId includes a HMAC of some additional information about the
+    // user-agent, as a simple way to check it's from the right place without
+    // having to share state between the content-server and auth-server.
+    var flowSignature = new Buffer(metadata.flowId.substr(32, 64), 'hex')
```

Comment:
Are there good semantic names we can give to the constants `0`, `32` and `64` in this block? To a crypto novice like myself they look a bit magical as it stands.